course/models.py

```diff
@@ -496,6 +489,64 @@ def default_language(self):
             return language_code
         return settings.LANGUAGE_CODE.split('-', 1)[0]
 
+    @property
+    def students(self):
+        return UserProfile.objects.filter(
+            enrollment__role=Enrollment.ENROLLMENT_ROLE.STUDENT,
+            enrollment__status=Enrollment.ENROLLMENT_STATUS.ACTIVE,
+            enrollment__course_instance=self
+        )
+
+    @property
+    def all_students(self):
+        return UserProfile.objects.filter(
+            enrollment__role=Enrollment.ENROLLMENT_ROLE.STUDENT,
+            enrollment__course_instance=self
+        ).annotate(enrollment_status=F('enrollment__status'))
+
+    @property
+    def assistants(self):
+        return UserProfile.objects.filter(
+            enrollment__role=Enrollment.ENROLLMENT_ROLE.ASSISTANT,
+            enrollment__status=Enrollment.ENROLLMENT_STATUS.ACTIVE,
+            enrollment__course_instance=self)
+
+    @property
+    def teachers(self):
+        return UserProfile.objects.filter(
+            enrollment__role=Enrollment.ENROLLMENT_ROLE.TEACHER,
+            enrollment__status=Enrollment.ENROLLMENT_STATUS.ACTIVE,
+            enrollment__course_instance=self)
+
+    @property
+    def course_staff(self):
+        return UserProfile.objects.filter(
+            Q(enrollment__role=Enrollment.ENROLLMENT_ROLE.TEACHER)
+            | Q(enrollment__role=Enrollment.ENROLLMENT_ROLE.ASSISTANT),
+            enrollment__status=Enrollment.ENROLLMENT_STATUS.ACTIVE,
+            enrollment__course_instance=self)
+
```

Comment:
The point of these convenience properties is that they return UserProfile instances instead of Enrollment instances. That is why the queries begin with "UserProfile.objects". I was under the impression that if you start the query with "self.enrollment_set", the queryset would return Enrollment instances (I spent quite a while Googling this...). If you know otherwise, I'd be glad if you could enlighten me.

Reply:
Oh, I didn't realize that it was querying UserProfile instead of Enrollment. My bad.